webapp/src/store/attachments.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSlice, PayloadAction} from '@reduxjs/toolkit'
+
+import {AttachmentBlock} from '../blocks/attachmentBlock'
+
+import {loadBoardData, initialReadOnlyLoad} from './initialLoad'
+
+import {RootState} from './index'
+
+type AttachmentsState = {
+    attachments: {[key: string]: AttachmentBlock}
+    attachmentsByCard: {[key: string]: AttachmentBlock[]}
+}
+
+const attachmentSlice = createSlice({
+    name: 'attachments',
+    initialState: {attachments: {}, attachmentsByCard: {}} as AttachmentsState,
```

Comment:
Yes, it is needed, we are using it in `cardDialog.tsx` to fetch all the attachments for a particular card.

Reply:
Actually, we use `attachmentsByCard` in `cardDetail.tsx`, but appears we need to keep it around for delete below...

I don't like trying to keep two objects in sync, it can lead to bugs. Would prefer to have a single source of truth.